scripts/maintenance/generate_audit_summary.py

```
@@ -0,0 +1,94 @@
+    except subprocess.CalledProcessError as e:
+        # Don't log errors for 'which' checks
+        if args[0] != 'which':
+            print(f"Error: {e.stderr}", file=sys.stderr)
+        return None
+
+def main():
+    if not os.path.exists('audit_results.json'):
+        print("Error: audit_results.json not found.", file=sys.stderr)
+        sys.exit(1)
+
+    with open('audit_results.json', 'r') as f:
```

Ruff (0.15.12)

[warning] Unnecessary mode argument

Remove mode argument

(UP015)